### DISCUSSION TITLE
Long translations end up switching back to English after 20 minutes

### DISCUSSION THREAD

**Discussion by chowder3907** [openai/whisper#667] [Q&A] [answered]
I am trying to use whisper to translate some longer videos (about an hour long ) to Spanish however after around the 20 minute mark, they end up not translating and only transcribing. After another ~20 minutes, it goes back to writing in Spanish. I'm not sure what I'm doing wrong. This is across multiple different videos and a consistent problem

**Comment by jongwook** [comment, marked as answer]
The models are not trained to transcribe into non-English languages, so it's actually not an intended use case (related: https://github.com/openai/whisper/discussions/649#discussioncomment-4325122). That being said, I guess in theory you could modify the decoding logic (the `update()` method below in particular`) to reject words or phrases in English.

https://github.com/openai/whisper/blob/0b5dcfdef7ec04250b76e13f1630e32b0935ce76/whisper/decoding.py#L199-L211
